src/states/node.rs

```diff
@@ -1593,6 +1616,72 @@ impl Node {
         result.with_value(Ok(()))
     }
 
+    fn handle_rt_req(&mut self,
+                     msg_id: MessageId,
+                     digest: sha256::Digest,
+                     src: Authority<XorName>,
+                     dst: Authority<XorName>)
+                     -> Evented<Result<(), RoutingError>> {
+        // TODO: Compare the hash and don't send anything if he requester is up-to-date.
+        let mut result = Evented::empty();
+        let sections = self.peer_mgr.pub_ids_by_section(self.full_id.public_id());
```

Comment:
Maybe the part about calculating the hash of the sections could be factored out into a method on `RoutingTable` or `PeerManager`? (It's duplicated in `handle_timeout`).